microsoft/botframework-solutions
# [Virtual Assistant Template][TypeScript] Add tests&code-coverage for the template

## Description
- Add new tests for Virtual Assistant
  - `MainDialog`
  - `Localization`
  - `Interruption`
  - `OnBoardingDialog`
- Implement `nyc` for **Code Coverage**
- Add `.nycrc` file
- Update `copier.js` to copy `nycrc` correctly
- Update `mocha.opts`
- Update `.gitignore`
- Update `sample-assistant`

## Testing Steps

1. Go to `AI\templates\Virtual-Assistant-Template\typescript\samples\sample-assistant`
2. Run `npm install`
3. Run `npm run coverage`
4. Check the tests of the V.A.Sample

![image](https://user-images.githubusercontent.com/12394167/57812813-87b86180-7744-11e9-8cbd-5f5932d4c14e.png)

5. Check the coverage of the V.A.Sample

![image](https://user-images.githubusercontent.com/12394167/57812654-14165480-7744-11e9-8ff5-89be9240d3b9.png)
